lib/storage/partition.go

```diff
@@ -1779,6 +1779,11 @@ func mustOpenParts(path string, partNames []string) []*partWrapper {
 	des := fs.MustReadDir(path)
 	m := make(map[string]struct{}, len(partNames))
 	for _, partName := range partNames {
+		partPath := filepath.Join(path, partName)
+		if !fs.IsPathExist(partPath) {
+			logger.Panicf("FATAL: missing part dir %q; part is listed in %q, but missing on disk", partPath, path+partsFilename)
```

Comment:
Should we add an action list to the error message? Like: "please check the backup was restored completely", or "verify there are no disk corruptions and restore from the backup", or "delete this part from parts.json if nothing helped"?

Reply:
Updated error message, ptal

> "please check the backup was restored completely"

this case is checked upon storage start and triggers respective error :)